### DISCUSSION TITLE
Connection errors: net::ERR_HTTP2_PROTOCOL_ERROR 200 (OK)

### DISCUSSION THREAD

**Discussion by scampos-smarttech** [featurehub-io/featurehub#1121] [Q&A]
Hi team, 
I am experiencing some connectivity issues. An error with code `net::ERR_HTTP2_PROTOCOL_ERROR 200 (OK)` is triggered when I leave the browser's tab of my web-app in second plane or if I leave the tab without interaction for long time. Sometimes (very few) It is triggered at the beginning of the connection, while I'm waiting for the connection to be `Readyness.Ready` and If the error is thrown, I have a delay of 30 seconds in the app initialization. Then the connection is recovered and start to work again.

Looking at network's tab in chrome's inspector, the "EventStream" works in three steps as follows:
1. Open the connection (ack)
2. Retrieve the flags data and
3. Closes the connection (bye) the connection and start over again in periods of 30 seconds.

But when the error happens, the event stream has the close step missing.
![Screenshot 2024-02-07 at 14 03 06](https://github.com/featurehub-io/featurehub/assets/156784176/7ea3cc3f-efca-4fcb-bde0-08338328f14f)
![Screenshot 2024-02-07 at 14 20 03](https://github.com/featurehub-io/featurehub/assets/156784176/b0e0bbd7-8c99-4961-9020-02b310a9f7b6)
![Screenshot 2024-02-07 at 14 21 04](https://github.com/featurehub-io/featurehub/assets/156784176/7e65d100-f3fe-419e-95b2-1c746d1794c0)

Do you know why this happens?
How can I handle these errors?

PS: This also happens to other people who use the application, so I rule out connection problems at home.

Thanks in advance!

**Comment by rvowles** [comment]
Do you have any anti-virus software in common? That typically interferes with it, but its a bit unusual to use streaming in a browser, are  you sure you  want to do that? Streaming generally  ties   up  one of the connections to the server  and if you don't need _immediate_ updates, polling every 30s is often more effective. 

The reconnect should keep going until it is explicitly told to stop however. Your server  isn't being overloaded? SSE holding open a connection means you will need an etra   Edge server for every  200 or so connections as well.

**Comment by rvowles** [comment]
Feel free to jump on our  slack and ask questions or DM me if you need confidential assistance.

**Comment by scampos-smarttech** [comment]
Hi, thanks for your response!
Yes, in my case I do have an antivirus on my machine, but I have not seen any reference in the log.
Your point about the type of connection seems very interesting to me, it will be worth trying the change. I will try to use the query system instead of the SSE connection.

Regarding error handling, is it possible to handle these errors from outside the SDK?

Thanks for your invitation, btw!
